quickwit/quickwit-indexing/src/actors/indexing_service.rs

```diff
@@ -478,22 +485,23 @@ impl IndexingService {
 
         let indexes_metadata_futures = new_pipeline_ids
             .iter()
-            .unique_by(|pipeline_id| pipeline_id.index_id.clone())
-            .map(|pipeline_id| self.index_metadata(ctx, &pipeline_id.index_id));
+            .unique_by(|pipeline_id| pipeline_id.index_config_id.clone())
+            .map(|pipeline_id| self.index_metadata(ctx, &pipeline_id.index_config_id.index_id));
         let indexes_metadata = try_join_all(indexes_metadata_futures).await?;
-        let indexes_metadata_by_index_id: HashMap<String, IndexMetadata> = indexes_metadata
+        let indexes_metadata_by_index_id: HashMap<IndexConfigId, IndexMetadata> = indexes_metadata
             .into_iter()
-            .map(|index_metadata| (index_metadata.index_config.index_id.clone(), index_metadata))
+            .map(|index_metadata| (index_metadata.index_config_id(), index_metadata))
             .collect();
 
         let mut failed_spawning_pipeline_ids: Vec<IndexingPipelineId> = Vec::new();
 
         // Add new pipelines.
         for new_pipeline_id in updated_pipeline_ids.difference(&running_pipeline_ids) {
             info!(pipeline_id=?new_pipeline_id, "Spawning indexing pipeline.");
+
             let index_metadata = indexes_metadata_by_index_id
-                .get(&new_pipeline_id.index_id)
-                .expect("`indexes_metadata_by_index_id` must contain the index ID.");
+                .get(&new_pipeline_id.index_config_id)
+                .expect("`indexes_metadata_by_index_id` must contain the index incarnation ID."); // TODO: should we now just warn here?
```

Comment:
Oops. I was planning to get back to that TODO but completely forgot with integration test yak shaving. 

> This is important, because we have a race condition that did not really exist before here.

That's what I thought when I added this TODO comment, but after thinking about it more today, it doesn't seem to me that it is a completely new race condition here. I think you get the same condition today if you create and immediately delete an index. The only difference in behavior between the current main branch and this PR can be triggered by also creating an index with the same name afterwards. In this case the current code gets confused and code in this PR continues to fail the same way as if the index wasn't re-created at all or the new index had a different name.

Anyway, a more important question here is should we fail the entire plan if one of the indices fail? Panic here seems like a nuclear option for a theoretically very possible scenario. So, I feel like simply ignoring this index might be a better solution. If I am reading the scheduler code correctly we are checking the applied plan against chitchat server, so any discrepancies should be reconcile on the next run.

Reply:
@fulmicoton I pushed some changes to improve readability and no longer panic here. Could you take a look?